src/timesync/timesyncd-manager.c

```diff
@@ -1290,3 +1449,154 @@ int bus_manager_emit_ntp_server_changed(Manager *m) {
 
         return 1;
 }
+
+#if ENABLE_TIMESYNC_NTS
+
+static int manager_nts_obtain_agreement(Manager *m) {
+        unsigned tls_patience_msec = m->nts_keyexchange_timeout_usec / USEC_PER_MSEC;
+
+        int r;
+        int socket = NTS_attach_socket(m->current_server_name->string, 4460, SOCK_STREAM);
+
+        log_debug("Performing key exchange with %s\n", m->current_server_name->string);
+
+        NTS_TLS *nts_handshake = NTS_TLS_setup(m->current_server_name->string, socket);
+        assert_return(nts_handshake, -ENOMEM);
```

Comment:
assert_return() should only be used to validate parameters of functions, and only if the caller is potentially not foreign, i.e. from some other codebase. or in other words: outside of libsystemd's public apis never use assert_return().